### PR TITLE
Add sitemap

### PR DESCRIPTION
**Pull request by nhsz**
This PR adds `sitemap.xml` to ESP site

## Description

- adds `next-sitemap` package to generate sitemaps
- updates `robots.txt`

### CODE REVIEW

File: next-sitemap.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  siteUrl: 'https://esp.ethereum.foundation',
+  exclude: [
+    '/applicants/grantee-finance/thank-you',
```
Comment (samajammin) [inline next-sitemap.js:4]: Also `/applicants/grantee-finance`

File: public/sitemap.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+<sitemap><loc>https://esp.ethereum.foundation/sitemap-0.xml</loc></sitemap>
```
Comment (samajammin) [inline public/sitemap.xml:3]: Why does this sitemap link to sitemap-0.xml? why not just have one file sitemap.xml with all our urls?

**Comment by samajammin** [comment]
@nhsz breaking build:
```
1:22:58 PM: error next-sitemap@2.5.7: The engine "node" is incompatible with this module. Expected version ">=14.18.3". Got "12.22.0"
1:22:58 PM: error Found incompatible module.
```
https://app.netlify.com/sites/ecosystem-support/deploys/622bbd72df39ee0008da5848

**Comment by pettinarip** [comment]
I've set up the latest lts node version (16.14.0).

File: next-sitemap.js (hunk 1 above)
Comment (nhsz) [reply next-sitemap.js:4]: It's already included, last item

File: public/sitemap.xml (hunk 2 above)
Comment (nhsz) [reply public/sitemap.xml:3]: This is the [index sitemap](https://github.com/iamvishnusankar/next-sitemap#index-sitemaps) generated automatically by the tool

**Comment by nhsz** [comment]
I'm using the latest 16.x node LTS too

**Review by pettinarip — APPROVED** [review]

File: public/robots.txt [hunk 3]
```diff
@@ -1,2 +1,5 @@
 User-agent: *
-Disallow: /api/
\ No newline at end of file
+Disallow: /api/
+
+Sitemap: https://esp.ethereum.foundation/sitemap.xml
+Sitemap: https://esp.ethereum.foundation/sitemap-0.xml
```
[the remark this answers is not in the document]
Comment (pettinarip) [reply public/robots.txt:5]: Cool, thanks!

**Pull request merged**